Content/Items/Forest/Armors.SlimePrinceMinion.cs

```diff
@@ -0,0 +1,357 @@
+﻿using StarlightRiver.Helpers;
+using System.Collections.Generic;
+using Terraria.Audio;
+using Terraria.Graphics.Effects;
+using Terraria.ID;
+
+namespace StarlightRiver.Content.Items.Forest
+{
+	public class SlimePrinceMinion : ModProjectile
+	{
+		public const int MAX_LIFE = 400;
+
+		public int life = MAX_LIFE;
+
+		public Vector2 startPoint;
+		public NPC target;
+
+		public Player Owner => Main.player[Projectile.owner];
+		public bool Merged => State == 3;
+
+		public ref float Timer => ref Projectile.ai[0];
+		public ref float State => ref Projectile.ai[1];
+		public ref float HealTimer => ref Projectile.ai[2];
+
+		public override string Texture => AssetDirectory.ForestItem + Name;
+
+		public override void SetStaticDefaults()
+		{
+			DisplayName.SetDefault("Slime Prince");
+		}
+
+		public override void SetDefaults()
+		{
+			Projectile.width = 32;
+			Projectile.height = 32;
+			Projectile.timeLeft = 2;
+			Projectile.minion = true;
+			Projectile.DamageType = DamageClass.Summon;
+			Projectile.aiStyle = -1;
+			Projectile.friendly = true;
+			Projectile.hostile = false;
+			Projectile.penetrate = -1;
+			Projectile.tileCollide = false;
+			Projectile.hide = true;
+		}
+
+		public override bool? CanHitNPC(NPC target)
+		{
+			return target == this.target && State == 1 && Timer > 30 && Timer < 90;
+		}
+
+		public override void AI()
+		{
+			Timer++;
+			HealTimer++;
+
+			var helm = Owner.armor[0].ModItem as SlimePrinceHead;
+
+			if (helm != null && helm.IsArmorSet(Owner.armor[0], Owner.armor[1], Owner.armor[2]))
+				Projectile.timeLeft = 2;
+
+			if (Vector2.Distance(Projectile.Center, Owner.Center) > 5000)
```

Comment:
good situation to have a netupdate = true, since it's rare and involves a drastic position change. even though its technically deterministic